### PR TITLE
Add snippet mixin for symbols

### PR DESCRIPTION
**Pull request by bitjammer**
Snippets need a place to put the chunks of its code listing.

Chunks contain source code, and an optional source language and
name for a subheading at the minimum.

### CODE REVIEW

**Comment by Kyle-Ye** [comment]
Could you please rebase it and add a new Symbol+Snippet.swift in the corresponding folder to follow the new organised style in this repo?

**Comment by Kyle-Ye** [comment]
Add my rebase suggestion here https://github.com/bitjammer/swift-docc-symbolkit/pull/1

File: Sources/SymbolKit/SymbolGraph/Misc/LineList.swift [hunk 1]
```diff
@@ -92,7 +92,7 @@ extension SymbolGraph {
         /// The lines making up this line list.
         public var lines: [Line]
 
-        init(_ lines: [Line]) {
+        public init(_ lines: [Line]) {
```
Comment (Kyle-Ye) [inline Sources/SymbolKit/SymbolGraph/Misc/LineList.swift:95]: Why we need LineList public here? I did not see any usage in the #10 and https://github.com/apple/swift-docc/pull/61/

Comment (bitjammer) [reply Sources/SymbolKit/SymbolGraph/Misc/LineList.swift:95]: This PR is also needed in https://github.com/apple/swift-package-manager/pull/3953.

**Comment by bitjammer** [comment]
Rebased. Thank you @Kyle-Ye I extracted the extension into a separate file.

**Comment by Kyle-Ye** [comment]
Maybe the new file added should be named as "Symbol+Snippet.swift" rather than "Symbol+SymbolGraph.swift".

Besides that, LGTM.

**Review by Kyle-Ye — CHANGES_REQUESTED** [review]
Maybe the new file added should be named as "Symbol+Snippet.swift" rather than "Symbol+SymbolGraph.swift".

Besides that, LGTM.

**Comment by bitjammer** [comment]
Hm, strange, that is what I thought I had named it.

**Comment by bitjammer** [comment]
Fixed that, thanks!

**Review by Kyle-Ye — APPROVED** [review]

**Comment by bitjammer** [comment]
@swift-ci Please test

**Comment by Kyle-Ye** [comment]
The PR add a public init interface for SymbolGraph.LineList and SymbolGraph.LineList.Line for reuse them in https://github.com/apple/swift-package-manager/pull/3953.

Should we accept this kind of usage? cc @franklinsch

**Comment by franklinsch** [comment]
I don't see an issue with that, it seems reasonable to expose public initializers for the SymbolKit's models.

**Pull request merged**